### ISSUE TITLE
conda create does not lock anything

### ISSUE THREAD

**Issue by epruesse** [conda/conda#7222]
**I'm submitting a...**
  - [x] bug report
  - [ ] feature request

## Current Behavior

Conda apparently does not make use of locks when modifying environments or package cache.

### Steps to Reproduce

```
conda create -n env1 --file env.txt &
conda create -n env2 --file env.txt &
conda create -n env3 --file env.txt
```

## Expected Behavior

Parallel invokation of `conda` commands should be handled in some sensible fashion. The simplest may be outright denial to have more than one `conda` instance operating on the same conda installation. Neat would of course be fine-grained locking of indices, package-cache entries and environments with conda instances neatly waiting on one another.

Curiously, there is a module `conda.lock`, but it does not appear to be used by e.g. `conda create`? Was locking perhaps previously in place but lost at some point?

**Comment by kalefranz** [comment]
Duplicate of #6742

**Comment by epruesse** [comment]
Sorry, didn't see that one. Thanks!
